### DISCUSSION TITLE
Filter on computed link on computed link

### DISCUSSION THREAD

**Discussion by tchak** [geldata/gel#3498] [Q&A] [answered]
I have the following schema:

```
module default {
  type Version {
    multi link fields -> Field {
      property position -> int32 {
        default := 0;
      }
    }
    multi link ordered_fields := (select .fields order by @position);
    multi link sections := (select .ordered_fields filter .type = 'section');
  }

  type Field {
    required property name -> str;
    required property type -> str;
    multi link versions := .<fields[is Version];
  }
}
```

I am getting the following error on migrate:

```
error: object type 'default::Field' has no link or property 'type'
    ┌─ ./dbschema/default.esdl:174:59
    │
174 │     multi link sections := (select .ordered_fields filter .type = 'section');
    │                                                           ^^^^^ error
```

I don't know if I am doing something wrong or if this is a bug. I have the same issue with a filter on an already filtered computed. I am also pretty sure I have something similar working on single computed links.

**Comment by elprans** [comment, marked as answer]
It's an SDL->DDL ordering bug.  SDL processor fails to notice that `sections` depends on `Field.type` and tries to declare it too early.

**Comment by tchak** [comment]
Ah, cool. Changing declaration order solved the problem 🎉 

I have another question. I added another link property and am trying to filter `sections` by it. I am getting the error `Field` has no property `private`. I guess link properties are not propagated on computed links implicitly. Is there a way to do it?

```
module default {
  type Field {
    required property name -> str;
    required property type -> str;
    multi link versions := .<fields[is Version];
  }

  type Version {
    multi link fields -> Field {
      property position -> int32 {
        default := 0;
      }
      property private -> bool;
    }
    multi link ordered_fields := (select .fields order by @position);
    multi link sections := (select .ordered_fields filter .type = 'section');
    multi link private_sections := (select .sections filter @private = true);
  }
}
```

**Reply by elprans** [reply]
The property thing is a bug/unimplemented feature.  Currently, link properties on unions of links (which are created when you do a backlink) are lost.  We're aware of the issue and the fix should appear soon.
